File: api/taskfy/src/main/java/com/taskfy/core/domain/tasks/model/Folder.java
package com.taskfy.core.domain.tasks.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PastOrPresent;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@Builder
@Getter
@Setter
@Table(name = "folder")
public class Folder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @NotNull(message = "User ID cannot be null")
    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "parent_id")
    private Long parentId;

    @NotBlank(message = "Folder name cannot be empty")
    @Size(max = 255, message = "Folder name cannot be longer than 255 characters")
    @Column(name = "name", nullable = false)
    private String name;

    @PastOrPresent(message = "Creation date must be in the past or present")
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @PastOrPresent(message = "Update date must be in the past or present")
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: taskfy_web_app/src/model/folder.js
class Folder {
    constructor({
      id = undefined,
      user_id,
      parent_id,
      name,
      created_at = new Date().toISOString(),
      updated_at = created_at,
    }) {
      this.id = id;
      this.user_id = user_id;
      this.parent_id = parent_id;
      this.name = name;
      this.created_at = created_at; // Se 'created_at' não for fornecido, usa o valor padrão
      this.updated_at = updated_at;
    }
  }
  
  export default Folder;
